utils/handlers/config_crud.go

```diff
@@ -0,0 +1,49 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/spaceuptech/space-cloud/modules/crud"
+	"github.com/spaceuptech/space-cloud/utils/admin"
+)
+
+func HandleGetConnectionState(adminMan *admin.Manager, crud *crud.Module) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// Get the JWT token from header
+		tokens, ok := r.Header["Authorization"]
+		if !ok {
+			tokens = []string{""}
+		}
+		token := strings.TrimPrefix(tokens[0], "Bearer ")
+
+		// Check if the request is authorised
+		if err := adminMan.IsTokenValid(token); err != nil {
+			w.WriteHeader(http.StatusUnauthorized)
+			json.NewEncoder(w).Encode(map[string]string{"error": err.Error()})
+			return
+		}
+
+		// Create a context of execution
+		ctx, cancel := context.WithTimeout(r.Context(), 5*time.Second)
+		defer cancel()
+
+		vars := mux.Vars(r)
+		dbType := vars["dbType"]
+
+		connState, err := crud.GetConnectionState(ctx, dbType)
+
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			json.NewEncoder(w).Encode(map[string]string{"error": err.Error()})
+			return
+		}
+		w.WriteHeader(http.StatusOK) //http status code
```

Comment:
if connState is true the set the headers to statusok & send empty object
if connState is false the set the headers to internalservererror & send empty object

Reply:
We shouldn't do this, let's return the bool value to the client and set the header as OK in either case.

Reply:
yes, @Jayesh333 told to do the above change

Reply:
rushikesh discard this change